src/modelgauge/suts/together_client.py

```diff
@@ -257,6 +266,75 @@ def translate_response(self, request: TogetherChatRequest, response: TogetherCha
         return SUTResponse(text=text, top_logprobs=logprobs)
 
 
+@modelgauge_sut(
+    capabilities=[
+        AcceptsTextPrompt,
+        AcceptsChatPrompt,
+        ProducesPerTokenLogProbabilities,
+    ]
+)
+class TogetherDedicatedChatSUT(TogetherChatSUT):
+    """A SUT based on dedicated Together endpoint. Supports automatic endpoint spin-up."""
+
+    _ENDPOINTS_URL = "https://api.together.xyz/v1/endpoints"
+
+    def __init__(self, uid: str, model: str, api_key: TogetherApiKey):
+        super().__init__(uid, model, api_key)
+        self.endpoint_id = self._get_endpoint_id()
+        self.endpoint_status = self._get_endpoint_status()
+
+    def _get_endpoint_id(self) -> str:
+        headers = {"accept": "application/json", "authorization": f"Bearer {self.api_key}"}
+        response = _retrying_request(self._ENDPOINTS_URL, headers, {}, "GET")
+        for endpoint in response.json()["data"]:
+            if endpoint["name"] == self.model:
+                return endpoint["id"]
+        raise APIException(f"No endpoint found for model {self.model}")
+
+    def _get_endpoint_status(self) -> str:
+        headers = {"accept": "application/json", "authorization": f"Bearer {self.api_key}"}
+        response = _retrying_request(f"{self._ENDPOINTS_URL}/{self.endpoint_id}", headers, {}, "GET")
+        return response.json()["state"]
+
+    def _spin_up_endpoint(self):
+        # Get latest endpoint status
+        self.endpoint_status = self._get_endpoint_status()
+        if self.endpoint_status == "STARTED":
+            return
+        elif self.endpoint_status in ["PENDING", "STOPPING"]:
+            # Wait and retry.
+            time.sleep(2*60) # 2 minutes
+            self._spin_up_endpoint()
+        elif self.endpoint_status in ["STARTING"]:
+            # Wait and retry.
+            time.sleep(8*60) # 8 minutes.
```

Comment:
So if we start something and it isn't ready in 8 minutes we wait another 8 minutes? Seems like a more granular retry would be better, yes?

Reply:
Yeah, 1 or 2. I spend a fair bit of time waiting for evaluators to start, so anything that gets that down is great by me.